### PR TITLE
Add global state accessor of jobs

### PR DESCRIPTION
**Pull request by ffbin**
## Why are these changes needed?
GCS Service job info handler uses table storage instead of redis accessor.
Python code needs to query gcs table info and get it from redis directly before, but now it can only get gcs table info from gcs service, so we add global state accessor for python code.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by raulchen** [comment]
@edoakes Can you also take a look at this PR? 
There will be a couple of PRs that migrate `state.py` to C++. The reason why we need this is because GCS's storage will be plugable. So Python should use C++'s `GcsClient` instead of directly talking to Redis.
This is the first PR that migrates job api.

File: src/ray/gcs/gcs_client/global_state_accessor.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+#ifndef RAY_GCS_GLOBAL_STATE_ACCESSOR_H
+#define RAY_GCS_GLOBAL_STATE_ACCESSOR_H
+
+#include "service_based_gcs_client.h"
+
+namespace ray {
+namespace gcs {
+
+class GlobalStateAccessor {
```
Comment (raulchen) [inline src/ray/gcs/gcs_client/global_state_accessor.h:23]: Document this class:
```
`GlobalStateAccessor` is used to provide synchronous interfaces to access data in GCS for the language front-end (e.g., Python's `state.py`).
```

File: src/ray/gcs/gcs_client/global_state_accessor.cc [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+bool GlobalStateAccessor::Connect() { return gcs_client_->Connect(*io_service_).ok(); }
+
+void GlobalStateAccessor::Disconnect() {
+  gcs_client_->Disconnect();
+  io_service_->stop();
+  thread_io_service_->join();
+}
+
+std::vector<std::string> GlobalStateAccessor::GetJobTable() {
```
Comment (raulchen) [inline src/ray/gcs/gcs_client/global_state_accessor.cc:54]: We can just return `vector<JobTableData>` here. And in Cython, we can parse C++ `JobTableData` to the final Python dict.

File: src/ray/gcs/gcs_client/global_state_accessor.h [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ public:
+  explicit GlobalStateAccessor(const std::string &redis_address,
+                               const std::string &redis_password, bool is_test = false);
+
+  bool Connect();
+
+  void Disconnect();
+
+  std::vector<std::string> GetJobTable();
```
Comment (raulchen) [inline src/ray/gcs/gcs_client/global_state_accessor.h:32]: better name this `GetAllJobInfo`.

File: src/ray/gcs/gcs_client/global_state_accessor.h (hunk 1 above)
Comment (ffbin) [reply src/ray/gcs/gcs_client/global_state_accessor.h:27]: fixed

File: src/ray/gcs/gcs_client/global_state_accessor.h (hunk 3 above)
Comment (ffbin) [reply src/ray/gcs/gcs_client/global_state_accessor.h]: fixed

File: src/ray/gcs/gcs_client/global_state_accessor.cc (hunk 2 above)
Comment (ffbin) [reply src/ray/gcs/gcs_client/global_state_accessor.cc]: Now gcs_utils.JobTableData.FromString can deserialize string of JobTableData directly, which is easier.

**Review by raulchen — APPROVED** [review]

**Pull request merged**